dali/pipeline/operators/sequence/sequence_rearrange.cu

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <utility>
+#include <vector>
+
+#include "dali/pipeline/operators/sequence/sequence_rearrange.h"
+
+namespace dali {
+
+template <>
+void SequenceRearrange<GPUBackend>::RunImpl(DeviceWorkspace *ws, const int idx) {
+  const auto &input = ws->Input<GPUBackend>(idx);
+  auto &output = ws->Output<GPUBackend>(idx);
+
+  std::vector<std::vector<Index>> new_list_shape;
+  std::vector<Index> list_elements_sizes;
+
+  for (decltype(input.ntensor()) tensor_idx = 0; tensor_idx < input.ntensor(); tensor_idx++) {
+    // TODO(klecki) not sure if we should allow for different length of sequence
```

Comment:
I think all readers assume now that sequence length is constant the same way as the batch size.